### PR TITLE
Make token ByteString

### PR DESCRIPTION
**Pull request by phadej**

### CODE REVIEW

File: src/GitHub/Request.hs [hunk 1]
```diff
@@ -256,8 +256,8 @@ makeHttpRequest auth r = case r of
     setAuthRequest _                                  = id
 
     getOAuthHeader :: Auth -> RequestHeaders
-    getOAuthHeader (OAuth token) = [("Authorization", BS8.pack ("token " ++ token))]
-    getOAuthHeader _                   = []
+    getOAuthHeader (OAuth token) = [("Authorization", "token " <> token)]
+    getOAuthHeader _             = []
```
Comment (zudov) [inline src/GitHub/Request.hs]: Not sure if that's a problem, but shouldn't it add the token in case of `EnterpriseOAuth`?

Comment (zudov) [inline src/GitHub/Request.hs]: It [seems](https://github.com/phadej/github/search?utf8=✓&q=EnterpriseOAuth) that `EnterpiseOAuth` is only used to provide an endpoint.

Admittedly, I know nothing about enterprise installations.

Also, according to [GitHub's docs](https://help.github.com/enterprise/2.4/admin/guides/user-management/authenticating-users-for-your-github-enterprise-instance/)

> Note: User authentication using GitHub OAuth is deprecated and can not be used for new installations.

Comment (phadej) [inline src/GitHub/Request.hs]: @zudov good spot, I have no Enterprise installation to test against, so I just leave this as it is.

Comment (phadej) [inline src/GitHub/Request.hs]: @zudov the deprecated authentication is to use **github.com** (i.e. this site) to authenticate in `github.yourcompany.com` enterprise installation.

**Comment by zudov** [comment]
Looks good

**Pull request merged**